### PR TITLE
Support YAML metadata (Issue #2801)

### PR DESCRIPTION
**Pull request by ralsina**
First step in #2801 : Naïve support for YAML metadata.

- Optional yaml import
- Fail only if required by posts
- Parse just like our traditional metadata

### CODE REVIEW

File: nikola/post.py [hunk 1]
```diff
@@ -1023,6 +1027,18 @@ def _get_metadata_from_file(meta_data):
     if not meta_data[0]:
         meta_data = meta_data[1:]
 
+    # If 1st line is '---', then it's YAML metadata
+    if meta_data[0] == '---':
+        if yaml is None:
+            utils.req_missing('pyyaml', 'use YAML metadata', optional=True)
+            raise ValueError('Error parsing metadata')
+        idx = meta_data.index('---', 1)
+        meta = yaml.load('\n'.join(meta_data[1:idx]))
```
Comment (Kwpolska) [inline nikola/post.py]: Always use `yaml.safe_load`.

Comment (ralsina) [inline nikola/post.py]: Ok

**Comment by felixfontein** [comment]
Shouldn't you also adjust the `split_metadata` function in `PageCompiler`? The new YAML (and TOML?) metadata allows empty lines between starting and ending `---` (resp. `+++`?).

**Comment by ralsina** [comment]
@felixfontein that's why this is a naive implementation, I don't want to spend two days thinking of corner cases :-)

**Pull request merged**